pep-9999.rst

```diff
@@ -13,9 +13,47 @@ Post-History:
 Abstract
 ========
 
+Optional type hints were introduced to the Python language in PEP 484
+[#pep484]_, based on the function annotation syntax from PEP 3107
+[#pep3107]_. PEP 484 also introduced *type stubs*, also called *stub files*,
+that provide type information for untyped Python packages and modules. Type
+stubs serve multiple purposes:
+
+* They are the only way to add type information to extension modules.
+* They can provide type information for packages that do not wish to
+  add them inline.
+* They can be distributed separately from the implementation.
+  This allows stubs to be develped at a different pace or by different
+  authors, which is especially useful when adding type annotations to
+  existing packages.
+* They act as documentation, succinctly explaining the external
+  API of a package, without including the implementation or private
+  members.
+
+This PEP aims to give guidance to both authors of type stubs and developers
+of type checkers. It describes the contents of compliant type stubs,
+suggests a style guide for them, and lists constructs that compliant type
+checkers are expected to support.
+
+A type stub is considered to be compliant if it only uses constructs decribed
+in this PEP. Type stub authors can elect to use additional constructs, but
+must be prepared that some type checkers will not parse them as expected.
+
+A type checker is considered to be compliant if
+it accepts all compliant type stubs. This means that it
+will parse such a document without error and does not interpret any
+construct in a contradictory manner, but a type checker is not
```

Comment:
Does this mean a type checker may parse some construct used in a stub file and ignore that construct, and still be considered compliant? i.e. as long as the type checker doesn't throw errors or misuse constructs listed in this PEP, it's compliant?

Reply:
That was the idea and that's how I interpreted the discussion in issue #6. Otherwise I think it would be very hard to achieve compliance, since we plan describe many advanced features here that I think not even mypy or pytype fully support at the moment. This also leaves the door open for "partial" type checkers (for example as part of a linter) or more lenient type checkers, like the one in PyCharm.

On the other hand, maybe it does make sense to distinguish between compliant and "complete" (better word needed) type checkers?

Reply:
I think the idea is fine, as we discussed in #6. I just think this section's wording is confusing.

I think a good fix would be:
- choose a phrase to summarize "constructs that compliant type checkers are expected to support" (from the 3rd paragraph). I think "standard constructs" works well.
- change this paragraph to something like:
```
A type checker is considered to be compliant if it accepts all compliant type stubs. This means it will parse a compliant type stub without error and does not interpret any standard construct in a contradictory manner. A compliant type checker is not required to interpret all standard constructs, and additionally a type checker can support extra constructs not described in this PEP. All deviations from this PEP should be documented.
```

We could distinguish between "compliant type checkers" and "fully-compliant type checkers," but I'm not sure that's necessary.

Reply:
Ah, sorry I misunderstood you. I was not really content with the last paragraph, and your version is a much-needed improvement.